frontend/src/app/coworking/room-reservation/reservation-table.service.ts

```diff
@@ -85,19 +96,47 @@ export class ReservationTableService {
     );
   }
 
+  getRoomInformation(): Observable<RoomDetails[]> {
+    return this.http.get<RoomDetails[]>(`/api/coworking/rooms/`);
+  }
+
   draftReservation(
     reservationsMap: Record<string, number[]>,
-    operationStart: Date
-  ): Observable<Reservation> {
-    const selectedRoom: { room: string; availability: number[] } | null =
-      this._findSelectedRoom(reservationsMap);
+    operationStart: Date,
+    users: (PublicProfile | Profile)[]
+  ) {
+    return this.publicToProfile(users).pipe(
+      switchMap((detailedUsers: Profile[]) => {
+        const selectedRoom = this._findSelectedRoom(reservationsMap);
+        if (!selectedRoom) {
+          return throwError(() => new Error('No room selected'));
+        }
 
-    if (!selectedRoom) throw new Error('No room selected');
-    const reservationRequest: ReservationRequest = this._makeReservationRequest(
-      selectedRoom!,
-      operationStart
+        return this.getRoomInformation().pipe(
+          map((rooms) => {
+            const room = rooms.find((room) => room.id === selectedRoom.room);
+            if (!room) {
+              throw new Error('Room not found');
+            }
+            if (users.length < room.capacity) {
+              throw new Error('Not enough people to reserve');
+            }
```

Comment:
Did you check for max capacity? On localhost, I tried reserving a room w/ capacity 2 with 3 users, and it succeeded.

Reply:
Hey Ajay, just skimming the CR's before I tackle them tomorrow,but for this one, I didn't check for more than the capacity because for example, if a group of 3 wants to use one of the pair programming rooms, I assumed that'd be fine. But if we want it to where it has to be exactly the capacity, I can change that easily!

Reply:
Sounds good, can loop in @KrisJordan for this one!